### PR TITLE
Add an Interface

### PR DESCRIPTION
**Pull request by Beakerboy**
I will use this to ensure that any object used in a Object matrix has the necessary methods to work correctly.

### CODE REVIEW

File: src/Number/MatrixObjectInterface.php [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?php
+namespace MathPHP\Number;
+
+interface MatrixObjectInterface
```
Comment (markrogoyski) [inline src/Number/MatrixObjectInterface.php]: How about renaming this something like:
- MatrixObjectOperations
- MatrixObjectArithmetic

File: src/Number/MatrixObjectInterface.php [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+{
+    public function add($object);
```
Comment (markrogoyski) [inline src/Number/MatrixObjectInterface.php]: The object parameters should also implement the interface, correct? If so, add the interface type declaration to the parameter.

**Comment by markrogoyski** [comment]
Thanks for the pull request. This makes sense to have objects implement the interface and then you can use the interface type declaration to ensure compatibility.

File: src/Number/MatrixObjectInterface.php (hunk 1 above)
Comment (Beakerboy) [inline src/Number/MatrixObjectInterface.php]: You choose it and I'll change it.

File: src/Number/MatrixObjectInterface.php (hunk 2 above)
Comment (Beakerboy) [inline src/Number/MatrixObjectInterface.php]: Maybe I should rename it $object_or_scaler.

File: src/Number/MatrixObjectInterface.php (hunk 1 above)
Comment (markrogoyski) [inline src/Number/MatrixObjectInterface.php]: Eventually I imagine there may be more than add/multiply/subtract, so how about MatrixObjectOperations?

File: src/Number/MatrixObjectInterface.php (hunk 2 above)
Comment (markrogoyski) [inline src/Number/MatrixObjectInterface.php]: OK. In that case, when you add the PHP docBlock comments, you can add like:
```php
@param mixed $object_or_scalar Explanation goes here
```

File: src/Number/MatrixObjectInterface.php (hunk 1 above)
Comment (markrogoyski) [inline src/Number/MatrixObjectInterface.php]: Actually, isn't this interface going to determine if the objects themselves know how to handle arithmetic operations with other objects and/or scalars? The fact that these objects will be used in a matrix might be just a detail. Are there any other contexts where these objects might get used. And if yes, maybe a more general name, like ObjectOperations or ObjectArithmetic.

Comment (Beakerboy) [inline src/Number/MatrixObjectInterface.php]: That's an excellent point. ObjectArithmetic it is.

**Pull request merged**